rp2040-hal/src/spi.rs

```diff
@@ -217,14 +228,25 @@ impl<D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<Disabled, D, P, DS> {
     }
 
     /// Set format and datasize
-    fn set_format(&mut self, data_bits: u8, mode: Mode) {
+    fn set_format(&mut self, data_bits: u8, frame_format: FrameFormat) {
         self.device.sspcr0.modify(|_, w| unsafe {
-            w.dss()
-                .bits(data_bits - 1)
-                .spo()
-                .bit(mode.0.polarity == Polarity::IdleHigh)
-                .sph()
-                .bit(mode.0.phase == Phase::CaptureOnSecondTransition)
+            w.dss().bits(data_bits - 1).frf().bits(match &frame_format {
+                FrameFormat::MotorolaSpi(_) => 0x00,
+                FrameFormat::TexasInstrumentsSynchronousSerial => 0x01,
+                FrameFormat::NationalSemiconductorMicrowire => 0x10,
```

Comment:
There seem to be a missing enum in the PAC for `FRF`.
https://github.com/rp-rs/rp2040-pac/issues/74 created to track that.

Not blocking here but worth updating in the future.